examples/voice_send.cr

```diff
@@ -73,36 +76,37 @@ client.on_message_create do |payload|
 
     client.create_message(payload.channel_id, "Playing DCA file `#{filename}`.")
 
-    # The `parse` method reads the frames consecutively and passes them to the
-    # block.
-    parser.parse do |frame|
-      # For smooth audio streams Discord requires one packet every
-      # 20 milliseconds. The `timed_run` method measures the time it takes to
-      # run the block and then sleeps 20 milliseconds minus that time, ensuring
-      # accurate timing.
-      #
-      # When simply reading from DCA, the time it takes to read, process and
-      # send the frame is small enough that `timed_run` doesn't make much of a
-      # difference (in fact, some users report that it actually makes things
-      # worse). If the processing time is not negligibly slow because you're
-      # doing something else than DCA parsing, or because you're reading from a
-      # slow source, or for any other reason, then it is recommended to use
-      # `timed_run`. Otherwise, simply doing something like
-      # `sleep 20.milliseconds` may suffice.
-      Discord::VoiceClient.timed_run do
-        # Perform the actual sending of the frame to Discord.
-        voice_client.not_nil!.play_opus(frame)
-      end
+    # For smooth audio streams Discord requires one packet every
+    # 20 milliseconds. The `every` method measures the time it takes to run the
+    # block and then sleeps 20 milliseconds minus that time before moving on to
+    # the next iteration, ensuring accurate timing.
+    #
+    # When simply reading from DCA, the time it takes to read, process and
+    # send the frame is small enough that `every` doesn't make much of a
+    # difference (in fact, some users report that it actually makes things
+    # worse). If the processing time is not negligibly slow because you're
+    # doing something else than DCA parsing, or because you're reading from a
+    # slow source, or for any other reason, then it is recommended to use
+    # `every`. Otherwise, simply using a loop and `sleep`ing `20.milliseconds`
+    # each time may suffice.
+    Discord.every(20.milliseconds) do
+      frame = parser.next_frame(reuse_buffer: true)
+      break unless frame
+
+      # Perform the actual sending of the frame to Discord.
+      voice_client.not_nil!.play_opus(frame)
     end
 
-    # The above code can also equivalently be realised as:
-
-    # Discord.every(20.milliseconds) do
-    #   frame = parser.next_frame(reuse_buffer: true)
-    #   break unless frame
+    # Alternatively, the above code can be realised as the following:
     #
-    #   voice_client.not_nil!.play_opus(frame)
+    # parser.parse do |frame|
+    #   Discord::VoiceClient.timed_run do
```

Comment:
Please move `timed_run` to `Discord` and make it take the same argument the same way as `every`. You should use `timed_run` in the definition of `every` tbh.